pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TransactionsBase.java

```diff
@@ -432,4 +433,24 @@ protected void validateTopicName(String property, String namespace, String encod
             throw new RestException(Response.Status.PRECONDITION_FAILED, "Topic name is not valid");
         }
     }
+
+    protected CompletableFuture<Void> internalScaleTransactionCoordinators(int replicas) {
+        CompletableFuture<Void> completableFuture = new CompletableFuture<>();
+        validateSuperUserAccessAsync()
+                .thenCompose((ignore)-> namespaceResources().getPartitionedTopicResources()
+                        .updatePartitionedTopicAsync(SystemTopicNames.TRANSACTION_COORDINATOR_ASSIGN, p -> {
+                            if (p.partitions >= replicas) {
+                                throw new RestException(Response.Status.NOT_ACCEPTABLE,
+                                        "Number of transaction coordinators should "
+                                                + "be more than the current number of transaction coordinator");
+                            }
+                            return new PartitionedTopicMetadata(replicas);
+                        }).thenAccept(r -> completableFuture.complete(null)))
```

Comment:
We can remove the `.thenAccept(r -> completableFuture.complete(null)))`, just return a CompletableFuture in `updatePartitionedTopicAsync`?